rotkehlchen/tests/unit/globaldb/test_upgrades.py

```diff
@@ -506,6 +508,46 @@ def test_upgrade_v5_v6(globaldb: GlobalDBHandler):
         ).fetchone()[0] == 0
 
 
+@pytest.mark.parametrize('globaldb_upgrades', [[]])
+@pytest.mark.parametrize('custom_globaldb', ['v6_global.db'])
+@pytest.mark.parametrize('target_globaldb_version', [6])
+@pytest.mark.parametrize('reload_user_assets', [False])
+def test_upgrade_v6_v7(globaldb: GlobalDBHandler):
+    """Test the global DB upgrade from v6 to v7"""
+    # Check the state before upgrading
+    with globaldb.conn.read_ctx() as cursor:
+        # check that location_asset_mappings table is not present in the database
+        assert table_exists(
+            cursor=cursor,
+            name='location_asset_mappings',
+        ) is False
+
+    # execute upgrade
+    with ExitStack() as stack:
+        patch_for_globaldb_upgrade_to(stack, 7)
+        maybe_upgrade_globaldb(
+            connection=globaldb.conn,
+            global_dir=globaldb._data_directory / GLOBALDIR_NAME,  # type: ignore
+            db_filename=GLOBALDB_NAME,
+        )
+    assert globaldb.get_setting_value('version', 0) == 7
+    with globaldb.conn.read_ctx() as cursor:
+        # check that location_asset_mappings table is present in the database
+        assert table_exists(
+            cursor=cursor,
+            name='location_asset_mappings',
+            schema=DB_CREATE_LOCATION_ASSET_MAPPINGS,
+        ) is True
+        # check that expected number of mappings are migrated
+        assert cursor.execute(
+            'SELECT COUNT(*) FROM location_asset_mappings',
+        ).fetchone()[0] == 2098
+        # check that all the exchanges mappings are present
+        assert cursor.execute(
+            'SELECT COUNT(*) location FROM location_asset_mappings GROUP BY location;',
+        ).fetchall() == [(33,), (259,), (178,), (153,), (167,), (109,), (113,), (57,), (32,), (48,), (88,), (68,), (264,), (77,), (34,), (37,), (69,), (73,), (97,), (32,), (110,)]  # noqa: E501
```

Comment:
can you maybe create a mapping with `location -> amount` and do a query per location?